src/app/clusters/icd-management-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,139 @@
+        const auto enabledCommands = kEnabledCommands();
+
+        // Get UserActiveModeTriggerHint
+        BitMask<IcdManagement::UserActiveModeTriggerBitmap> userActiveModeTriggerHint(0);
+        if (Clusters::IcdManagement::Attributes::UserActiveModeTriggerHint::Get(endpointId, &userActiveModeTriggerHint) !=
+            Protocols::InteractionModel::Status::Success)
+        {
+            ChipLogError(Zcl, "Failed to get UserActiveModeTriggerHint, using default (0)");
+        }
+
+        // Get UserActiveModeTriggerInstruction
+        char instructionBuffer[kUserActiveModeTriggerInstructionMaxLength];
+        MutableCharSpan instructionSpan(instructionBuffer, sizeof(instructionBuffer));
+        CharSpan userActiveModeTriggerInstruction;
+
+        if (Clusters::IcdManagement::Attributes::UserActiveModeTriggerInstruction::Get(endpointId, instructionSpan) !=
+            Protocols::InteractionModel::Status::Success)
+        {
+            ChipLogError(Zcl, "Failed to get UserActiveModeTriggerInstruction, using default (empty string)");
+            userActiveModeTriggerInstruction = CharSpan();
+        }
+        else
+        {
+            userActiveModeTriggerInstruction = CharSpan(instructionSpan.data(), instructionSpan.size());
+        }
+
+        gServer.Create(Server::GetInstance().GetPersistentStorage(), *Server::GetInstance().GetSessionKeystore(),
+                       Server::GetInstance().GetFabricTable(), ICDConfigurationData::GetInstance().GetInstance(),
+                       BitFlags<IcdManagement::Feature>(featureMap), optionalAttributeSet, enabledCommands,
+                       userActiveModeTriggerHint, userActiveModeTriggerInstruction);
+        return gServer.Registration();
+    }
+
+    ServerClusterInterface & FindRegistration(unsigned clusterInstanceIndex) override { return gServer.Cluster(); }
+    void ReleaseRegistration(unsigned clusterInstanceIndex) override { gServer.Destroy(); }
+};
+} // namespace
+
+void MatterIcdManagementPluginServerInitCallback()
+{
+    PersistentStorageDelegate & storage           = Server::GetInstance().GetPersistentStorage();
+    Crypto::SymmetricKeystore * symmetricKeystore = Server::GetInstance().GetSessionKeystore();
+    FabricTable & fabricTable                     = Server::GetInstance().GetFabricTable();
+    ICDConfigurationData & icdConfigurationData   = ICDConfigurationData::GetInstance().GetInstance();
+
+    IntegrationDelegate integrationDelegate;
+    CodegenClusterIntegration::RegisterServer(
+        {
+            .endpointId                      = kRootEndpointId,
+            .clusterId                       = IcdManagement::Id,
+            .fixedClusterServerEndpointCount = IcdManagement::StaticApplicationConfig::kFixedClusterConfig.size(),
+            .maxEndpointCount                = 1, // only root-node functionality supported by this implementation
+            .fetchFeatureMap                 = true,
```

Comment:
Why?  We didn't use to allow setting the FeatureMap for this cluster via ZAP config; why would we start now?